index.js

```diff
@@ -1,102 +1,106 @@
-  function load() {
-    fetch(url + new URLSearchParams({
-      page: page,
-      per_page: per_page,
-    }))
-        .then(function (respose) {
-          return respose.json()
-        })
-        .then(function (responseData) {
-          total_count = responseData.total;
-          // Render total results.
-          total.innerHTML = `${per_page} of ${responseData.total} Records`;
-
-          let result = document.querySelector('#result');
-          // Clear result.
-          result.innerHTML = ""
-
-          // load responce data.
-          let data = responseData.data
-
-          let html = data.map(function(item) {
-            // here we need to generate html for object.
-            return `<div class="user-info"><div class="user-first-name">${item.first_name}</div><div class="user-last-name">${item.last_name}</div><div class="user-email">${item.email}</div><div class="user-avatar"><img src="${item.avatar}"></div></div>`
-          })
-
-          result.insertAdjacentHTML('afterbegin', html.join(''))
-        })
-  }
+async function load() {
+  const response = await fetch(pagination.url + new URLSearchParams({
+    page: pagination.page,
+    per_page: pagination.perPage,
+  }));
+  const responseData = await response.json();
+  pagination.totalCount = responseData.total;
+  // Render total results.
+  pagination.total.innerHTML = `${pagination.perPage} of ${responseData.total} Records`;
+
+  const result = document.querySelector('#result');
+  // Clear result.
+  result.innerHTML = ""
+
+  // load responce data.
+  const data = responseData.data
+  const html = data.map(function(item) {
+    // here we need to generate html for object.
+    return `<div class="user-info"><div class="user-first-name">${item.first_name}</div><div class="user-last-name">${item.last_name}</div><div class="user-email">${item.email}</div><div class="user-avatar"><img src="${item.avatar}"></div></div>`
+  })
+
+  result.insertAdjacentHTML('afterbegin', html.join(''))
+}
 
 
   function numPages() {
-    return Math.ceil(total_count / per_page);
+    return Math.ceil(pagination.totalCount / pagination.perPage);
   }
 
   // Set number of page.
-  function setPageNumber(page_number) {
-    page_number_span.innerHTML = page_number;
+  function setPageNumber(pageNumber) {
+    pagination.pageNumberSpan.innerHTML = pageNumber;
 
     // Hide Prev button if we are on the first page.
-    btn_prev.style.visibility = page_number === 1 ? 'hidden' : 'visible';
+    pagination.btnPrev.style.visibility = pageNumber === 1 ? 'hidden' : 'visible';
 
     // Hide Next button if we are on the last page.
-    btn_next.style.visibility = page_number === numPages() ? 'hidden' : 'visible';
+    pagination.btnNext.style.visibility = pageNumber === numPages() ? 'hidden' : 'visible';
   }
 
+  function setPageLoad() {
+    setPageNumber(pagination.pageNumber);
+    load();
+  }
 
+  function paginationClick(btnType) {
+    if (btnType === 'next') {
```

Comment:
I'd recommend to create separated variable for button types: `const BUTTON_TYPES = {`
`  PREV: 'prev', `
`  NEXT: 'next; `
`}`, and use it.